src/consumer/engine.rs

```diff
@@ -146,37 +148,59 @@ impl<Exe: Executor> ConsumerEngine<Exe> {
                     })?;
             }
 
-            if self.remaining_messages < self.batch_size.div_ceil(2) {
+            // debug!("self.remaining_messages ({}) < self.batch_size.div_ceil(2) ({}) = {}",
+            // self.remaining_messages, self.batch_size.div_ceil(2), self.remaining_messages <
+            // self.batch_size.div_ceil(2));
+            if self.remaining_messages < (self.batch_size.div_ceil(2) as i64) {
                 match self
                     .connection
                     .sender()
-                    .send_flow(self.id, self.batch_size - self.remaining_messages)
+                    .send_flow(
+                        self.id,
+                        (self.batch_size as i64 - self.remaining_messages) as u32,
```

Comment:
I don't know your attended behavior but if you cast a negative number to u32 this gives a big number

```
let a = 12_i64;
let b = 23_i64;
dbg!((a - b) as u32); // 4294967285
```

Reply:
the batch size is always positive as it is a u32 and the remaining messages could be negative, but in that case it will output a positive number that fit in u32 as batch size is the maximum value that we want to converge.

Reply:
Besides, if we ask of u32::MAX to the broker, it is not really an issue as well :P, it is better than receiving nothing